lnst/Recipes/ENRT/ConfigMixins/DevFlowsPinningHWConfigMixin.py

```diff
@@ -0,0 +1,236 @@
+"""
+Module containing mixin class for BaseEnrtRecipe class that enables flow
+steering across test device queues using steering rules.
+
+Copyright 2025 Red Hat, Inc.
+Licensed under the GNU General Public License, version 2 as
+published by the Free Software Foundation; see COPYING for details.
+"""
+
+__author__ = """
+sdobron@redhat.com (Samuel Dobron)
+"""
+
+
+from collections.abc import Iterable
+import re
+import logging
+from ipaddress import ip_address
+from typing import Literal, Optional, Any
+from lnst.Common.IpAddress import Ip4Address
+from lnst.Common.LnstError import LnstError
+from lnst.Devices.Device import Device
+
+from lnst.RecipeCommon.Perf.Measurements.BaseFlowMeasurement import Flow
+from lnst.Recipes.ENRT.BaseEnrtRecipe import EnrtConfiguration
+from lnst.Recipes.ENRT.ConfigMixins.BaseHWConfigMixin import BaseHWConfigMixin
+
+
+class DevFlowsPinningHWConfigMixin(BaseHWConfigMixin):
+    """
+        This class is an extension to the :any:`BaseEnrtRecipe` class that
+        enables flow steering across test device queues. Generated flows
+        are iterated and pinned to
+        :attr:`lnst.RecipeCommon.Perf.Measurements.BaseFlowMeasurement.Flow.receiver_cpupin`
+        CPU queue. Flow steering "keys" are configurable by overriding the
+        :meth:`.steer_flow_by` method and can be set independently.
```

Comment:
Could you explain this a bit more? The steer_flow_by() should be overriden by a derived class - how this would look like?

Reply:
Looks better with the docstring.